src/qibolab/transpilers/placer.py

```diff
@@ -0,0 +1,261 @@
+        self.connectivity = connectivity
+
+    def __call__(self, circuit: Circuit):
+        """Find the trivial placement for the circuit.
+
+        Args:
+            circuit (qibo.models.Circuit): Circuit model to check.
+        """
+        return dict(zip(list("q" + str(i) for i in range(circuit.nqubits)), range(circuit.nqubits)))
+
+
+class Custom(Placer):
+    """Define a custom initial qubit mapping.
+
+    Attributes:
+        map (list or dict): Physical to logical qubit mapping,
+        example [1,2,0] or {"q0":1, "q1":2, "q2":0} to assign the
+        physical qubits 0;1;2 to the logical qubits 1;2;0 respectively.
+        connectivity (networkx.Graph): chip connectivity.
+    """
+
+    def __init__(self, map, connectivity=None, verbose=False):
+        """Args:
+        map (list or dict): Physical to logical qubit mapping,
+        example [1,2,0] or {"q0":1, "q1":2, "q2":0} to assign the
+        physical qubits 0;1;2 to the logical qubits 1;2;0 respectively.
+        connectivity (networkx.Graph): chip connectivity.
+        verbose (Bool): if "True" print info messages.
+        """
+        self.connectivity = connectivity
+        self.map = map
+        self.verbose = verbose
+
+    def __call__(self, circuit=None):
+        """Return the custom placement if it can be applied to the given circuit (if given).
+
+        Args:
+            circuit (qibo.models.Circuit): Circuit to be transpiled.
+        """
+        if isinstance(self.map, dict):
+            pass
+        elif isinstance(self.map, list):
+            self.map = dict(zip(list("q" + str(i) for i in range(len(self.map))), self.map))
+        else:
+            raise_error(TypeError, "Use dict or list to define mapping.")
+        if circuit is not None:
+            if not check_placement(circuit, self.map, self.verbose):
+                raise_error(ValueError)
+        elif not check_mapping_consistency(self.map, self.verbose):
+            raise_error(ValueError)
+        return self.map
+
+
+class Subgraph(Placer):
+    """
+    Subgraph isomorphism qubit placer, NP-complete it can take a long time
+    for large circuits. This initialization method may fail for very short circuits.
+
+    Attributes:
+        connectivity (networkx.Graph): chip connectivity.
+    """
+
+    def __init__(self, connectivity):
+        """Args:
+        connectivity (networkx.graph): chip connectivity.
+        """
+        self.connectivity = connectivity
+
+    def __call__(self, circuit: Circuit):
+        """Find the initial layout of the given circuit using subgraph isomorphism.
+
+        Args:
+            circuit (qibo.models.Circuit): Circuit to be transpiled.
+        """
+        # TODO fix networkx.GM.mapping for small subgraphs
+        circuit_repr = create_circuit_repr(circuit)
+        if len(circuit_repr) < 3:
+            raise_error(ValueError, "Circuit must contain at least two two qubit gates to implement subgraph placement")
+        H = nx.Graph()
```

Comment:
```suggestion
        h = nx.Graph()
```
Try to stick to naming conventions (i.e. `snake_case` for variables), and, if possible, choose more telling names.